daphne_worker/src/durable/mod.rs

```diff
@@ -225,70 +234,86 @@ impl<'srv> DurableConnector<'srv> {
     }
 }
 
-macro_rules! ensure_garbage_collected {
-    ($req:expr, $object:expr, $id:expr, $binding:expr) => {{
-        if $req.path() == crate::durable::DURABLE_DELETE_ALL && $req.method() == Method::Post {
-            $object.state.storage().delete_all().await?;
-            $object.touched = false;
-            return Response::from_json(&());
-        } else if !$object.touched {
+/// Run garbage collection requests.
+///
+/// If a garbage collection request is handled no further processing needs to be done, as such,
+/// this function will consume the request, otherwise it will return the passed in request for
+/// further handling.
+async fn run_garbage_collection(
+    req: Request,
+    state: &State,
+    env: &Env,
+    deployment: crate::config::DaphneWorkerDeployment,
```

Comment:
* Suggestion: `staet`, `env`, `deployment`, and `touched` are attributes common to all DOs that are "garbage collected". WDYTA about defining a trait for these and have `run_garbage_collection` be a derived method on this trait? I think this would make the code a lot cleaner.
* This doesn't actually "run" garbage collection: It puts the DO instance on the garbage collection queue, which is dequeued before each e2e tests.

Reply:
To be clear: The second suggestion is to rename the method.